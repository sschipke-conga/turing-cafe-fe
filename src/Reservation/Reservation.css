.reservation {
  background: white;
  border: 5px solid teal;
  border-radius: 10px;
  color: teal;
  height: 45%;
  margin: 10px;
  width: 17%;
}

.reservation button {
  background: teal;
  color: white;
  height: 25px;
  margin-bottom: 10px;
  width: 80%;
}