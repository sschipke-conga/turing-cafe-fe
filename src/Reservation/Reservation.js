import React from 'react';
import './Reservation.css'

const Reservation = ({reservation, cancelReservation}) => {
  const {name, date, time, number, id} = reservation;
  return (
    <div className="reservation">
      <h2>{name}</h2>
      <p>{date}</p>
      <p>{time} pm</p>
      <p>Number of Guests: {number}</p>
      <button id={id} onClick={cancelReservation} type="button">Cancel</button>
    </div>
  )
}

export default Reservation