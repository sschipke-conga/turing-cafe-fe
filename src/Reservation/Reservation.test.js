import React from 'react';
import { shallow } from 'enzyme';
import Reservation from './Reservation'

describe('reservation', () => {
  let wrapper, mockReservation;
  const mockCancelReservation = jest.fn();
  beforeEach(() =>{
    mockReservation = {
      id: 123,
      name: 'Kimberly',
      date: '10/25',
      time: '8:00',
      number: 6
    }
    wrapper =  shallow(
      <Reservation reservation={mockReservation} cancelReservation={mockCancelReservation}/>
    )
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})