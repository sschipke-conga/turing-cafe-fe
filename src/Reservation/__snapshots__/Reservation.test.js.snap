// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reservation should match the snapshot 1`] = `
<div
  className="reservation"
>
  <h2>
    Kimberly
  </h2>
  <p>
    10/25
  </p>
  <p>
    8:00
     pm
  </p>
  <p>
    Number of Guests: 
    6
  </p>
  <button
    id={123}
    onClick={[MockFunction]}
    type="button"
  >
    Cancel
  </button>
</div>
`;
